lib/Doctrine/ORM/Query/AST/BetweenExpression.php

```diff
@@ -8,28 +8,18 @@
 
 class BetweenExpression extends Node
 {
-    /** @var ArithmeticExpression */
-    public $expression;
-
-    /** @var ArithmeticExpression */
-    public $leftBetweenExpression;
-
-    /** @var ArithmeticExpression */
-    public $rightBetweenExpression;
-
-    /** @var bool */
-    public $not;
+    public bool $not;
```

Comment:
```suggestion
    public bool $not = false;
```

We should not allow this property to be uninitialized. In fact, I'd suggest to add it as promoted property to the constructor. `Parser::BetweenExpression()` currently assigns it right after construction anyway.

See #10267.